Source/CronosPlayUnreal/Public/DefiWalletCoreActor.h

```diff
@@ -517,281 +535,236 @@ class CRONOSPLAYUNREAL_API ADefiWalletCoreActor : public AActor {
               Category = "CronosPlayUnreal")
     void Erc721TokenByIndex(FString contractAddress, FString erc721index,
-                            FString &token, bool &success,
-                            FString &output_message);
+                            FGeneralCronosCallDelegate Out);
 
     /**
      * Returns a token ID owned by owner at a given index of its token list. Use
      * along with balanceOf to enumerate all of owner's tokens.
      * @param contractAddress erc 721 contract address
      * @param erc721owner owner
      * @param erc721index which index
-     * @param token a token ID at a given index
-     * @param success whether succeed or not
-     * @param output_message error message, "" if succeed
+     * @param Out callback
      */
     UFUNCTION(BlueprintCallable,
               meta = (DisplayName = "Erc721TokenOwnerByIndex",
                       Keywords = "Wallet"),
               Category = "CronosPlayUnreal")
     void Erc721TokenOwnerByIndex(FString contractAddress, FString erc721owner,
-                                 FString erc721index, FString &token,
-                                 bool &success, FString &output_message);
+                                 FString erc721index,
+                                 FGeneralCronosCallDelegate Out);
 
     /**
      * Get erc-1155 uri
      * @param contractAddress erc1155 contract address
      * @param tokenID token ID
-     * @param uri  get uri
-     * @param success whether succeed or not
-     * @param output_message error message, "" if succeed
+     * @param Out callback
      */
     UFUNCTION(BlueprintCallable,
               meta = (DisplayName = "Erc1155Uri", Keywords = "Wallet"),
               Category = "CronosPlayUnreal")
-    void Erc1155Uri(FString contractAddress, FString tokenID, FString &uri,
-                    bool &success, FString &output_message);
+    void Erc1155Uri(FString contractAddress, FString tokenID,
+                    FGeneralCronosCallDelegate Out);
 
     /**
      * Get erc-1155 IsApprovedForAll
      * @param contractAddress erc1155 contract address
      * @param erc1155owner owner address
      * @param erc1155approvedaddress  approved address
-     * @param result is approved for all
-     * @param success whether succeed or not
-     * @param output_message error message, "" if succeed
+     * @param Out callback
      */
     UFUNCTION(BlueprintCallable,
               meta = (DisplayName = "Erc1155IsApprovedForAll",
                       Keywords = "Wallet"),
               Category = "CronosPlayUnreal")
     void Erc1155IsApprovedForAll(FString contractAddress, FString erc1155owner,
-                                 FString erc1155approvedaddress, bool &result,
-                                 bool &success, FString &output_message);
+                                 FString erc1155approvedaddress,
+                                 FGeneralCronosCallBoolDelegate Out);
```

Comment:
Same, better off assigning a dedicated delegate. Rename it.

Reply:
ok